src/seq/minhash.jl

```diff
@@ -0,0 +1,118 @@
+# MinHash
+# ========
+#
+# Functions to generate MinHash sketches of biological sequences
+#
+# This file is a part of BioJulia.
+# License is MIT: https://github.com/BioJulia/Bio.jl/blob/master/LICENSE.md
+using DataStructures: SortedSet
+
+"""
+MinHash Sketch type
+
+MinHash sketches are a sorted set of kmer hashes, containing the smallest `s`
+hashes for kmers of a given length. The type contains two parameters:
+
+* .sketch: a sorted set of hashes
+* .kmersize: the length of kmers used to generate the sketch
+"""
+type MinHashSketch
+    sketch::Vector{UInt64}
+    kmersize::Int
+
+    function MinHashSketch(sketch::Vector, kmersize::Int)
+        length(sketch) > 0 || error("Sketch cannot be empty")
+        kmersize > 0 || error("Kmersize must be greater than 0")
+        new(sketch, kmersize)
+    end
+end
+
+Base.getindex(s::MinHashSketch, part) = getindex(s.sketch, part)
+Base.length(s::MinHashSketch) = length(s.sketch)
+Base.size(s::MinHashSketch) = (length(s), s.kmersize)
```

Comment:
Is there something that would be better (something like `dims()`, or should I just not bother?